c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1106 @@
+/*
+ * Copyright (c) 2019-2020 The University of Manchester
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// ------------------------------------------------------------------------
+//
+// Extra definitions of things on SpiNNaker chips that aren't already mentioned
+// in spinnaker.h, or where the description is miserable.
+//
+// ------------------------------------------------------------------------
+
+#ifndef __SPINN_EXTRA_H__
+#define __SPINN_EXTRA_H__
+
+#include <spinnaker.h>
+#include <stdbool.h>
+#if defined(__GNUC__) && __GNUC__ < 6
+// This particular warning (included in -Wextra) is retarded wrong for client
+// code of this file. Only really a problem on Travis.
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif // __GNUC__
+
+// Generates valid code if the named type is one word long, and invalid code
+// otherwise. It would be simpler if we used _Static_assert, but Eclipse seems
+// to hate that.
+#define ASSERT_WORD_SIZED(type_ident) \
+    struct __static_word_sized_assert_ ## type_ident { \
+        char _dummy[2 * (sizeof(type_ident) == sizeof(uint)) - 1]; \
+    }
+
+// ---------------------------------------------------------------------
+// 1. Chip Organization
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 2. System Architecture
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 3. ARM968 Processing Subsystem
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 4. ARM 968
+
+// No special registers here
+
+// ---------------------------------------------------------------------
+// 5. Vectored Interrupt Controller
+
+typedef void (*vic_interrupt_handler_t) (void);
+
+typedef union {
+    struct {
+        uint watchdog : 1;
```

Comment:
should we be commeting these seperately?

Reply:
this covers all structs. as they all open up the question

Reply:
documented